export the debug logging facility

majewsky opened the pull request:
There's already a nice amount of debug logging in this module, which I could use to test its behavior while integrating it into [Limes](https://github.com/sapcc/limes), and to validate new policies more easily. But it's all private and there's no way to enable it!

This PR extends the `Context` struct such that I can plug my own logging facility into it if I wish to.

majewsky committed:
export the debug logging facility

databus23 commented:
Hmm, I think we need to update the travis.yaml to use a recent version of go and remove the go vet installation, its included in the go toolchain itself (`go vet`). @majewsky could you update this in PR to check if it fixes the build?

majewsky committed:
update Travis to use newer go's

It fails right now on not being able to install `go vet` anymore (since
it is included out of the box with newer go's.)

majewsky committed:
document the logging facility in the README

majewsky commented:
Done, looks better now.

databus23 commented:
LGTM, Thanks!
